spec/controllers/course/discussion/topics_controller_spec.rb

```diff
@@ -17,80 +17,125 @@
     end
     let(:topics) { controller.instance_variable_get(:@topics) }
 
-    before { sign_in(user) }
-
     describe '#index' do
       subject { get :index, course_id: course }
 
-      it { is_expected.to render_template(:index) }
+      context 'when a course staff visits the page' do
+        before { sign_in(staff) }
+
+        it { is_expected.to render_template(:index) }
+
+        it 'shows all topics' do
+          subject
+          expect(topics).to contain_exactly(topic, pending_topic)
+        end
+      end
+
+      context 'when a course student visits the page' do
+        before { sign_in(student) }
+
+        it { is_expected.to render_template(:index) }
 
-      it 'shows all topics' do
-        subject
-        expect(topics).to contain_exactly(topic, pending_topic)
+        it 'shows all topics' do
```

Comment:
In this case all topics are from that student, I will change the description to resolve confusion ><